File: Data Structure/Hash Tables/Hashing.py
hashTable = [[] for _ in range(5)]

def checkPrime(n):
    if n == 1 or n == 0:
        return 0

    for i in range(2, n // 2):
        if n % i == 0:
            return 0

    return 1


def getPrime(n):
    if n % 2 == 0:
        n = n + 1

    while not checkPrime(n):
        n += 2

    return n


def hashFunction(key):
    capacity = getPrime(10)
    return key % capacity


def insertValue(key, value):
    index = hashFunction(key)
    found = False
    for i, k in enumerate(hashTable[index]):
        if k[0] == key:
            hashTable[index][i] = (key, value) 
            found = True
            break
    if not found:
        hashTable[index].append((key, value)) 


def removeValue(key):
    index = hashFunction(key)
    for i, kv in enumerate(hashTable[index]):
        if kv[0] == key:
            del hashTable[index][i]
            break

insertData(213, "orange") 

removeData(123)

print(hashTable)

File: Data Structure/Hash Tables/Hashing.cpp
#include<iostream>
#include<list>

using namespace std;

class HashTable{
    int capacity;
    list<int> *table;

public:
    HashTable(int i):
        void insertValue(int key, int value);

};

    void HashTable::insertValue(int key, int value)
    {
        int index = hashFunction(key);
        table[index].push_back(value);
    }

    void HashTable::deleteValue(int key)
    {
        int index = hashFunction(key);
        list<int>::iterator i;
        for(i=table[index].end(); i++)
        {
            if(*i == key)
                break;
        }

        if(i!=table[index].end())
            table[index].erase(i);

    }


int main()
{
    int key[] = {1,2,3,4,5};
    int value[] = {11,22,33,44,55};

}




